### PR TITLE
Added FileHasher

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

File: neptune/new/internal/artifacts/file_hash.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        path = Path.home() / ".neptune"
+        os.makedirs(path, exist_ok=True)
+
+        return datalite(db_path=str(path / "files.db"))(inner_cls, *args, **kwargs)
+    return decorator(cls)
+
+
+@persist
+@dataclass
+class FileHash:
```
Comment (pkasprzyk) [inline neptune/new/internal/artifacts/file_hash.py:35]: LocalFileHash?

File: neptune/new/internal/artifacts/file_hash.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import typing
+from pathlib import Path
+from dataclasses import dataclass
+
+from datalite import datalite
```
Comment (pkasprzyk) [inline neptune/new/internal/artifacts/file_hash.py:21]: have you used that? I doesn't seem to have much traction (3 stars on GH...)

File: neptune/new/internal/artifacts/file_hasher.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                return stored_file_hash.file_hash
+            else:
+                computed_hash = sha1(absolute)
+                stored_file_hash.file_hash = computed_hash
+                stored_file_hash.modification_date = modification_date
+                stored_file_hash.update_entry()
+
+                return computed_hash
+
+        computed_hash = sha1(absolute)
```
Comment (pkasprzyk) [inline neptune/new/internal/artifacts/file_hasher.py:50]: could you wrap this logic in `else:`? I think the flow will be clearer then

File: tests/neptune/new/internal/artifacts/test_file_hasher.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        self.assertEqual("63d995a30adf77a40305ce7c69417866666d7df3", FileHasher.get_artifact_hash(reversed(artifacts)))
+
+    @patch('pathlib.Path.home')
+    def test_local_file_hash(self, home):
+        home.return_value = Path(self.temp.name)
+
+        self.assertEqual(
+            'd78f8bb992a56a597f6c7a1fb918bb78271367eb',
+            FileHasher.get_local_file_hash(f'{self.temp.name}/test')
+        )
```
Comment (pkasprzyk) [inline tests/neptune/new/internal/artifacts/test_file_hasher.py:71]: can you add validation for sqlite caching here? I don't think it's tested anywhere right now

**Review by pkasprzyk — CHANGES_REQUESTED** [review]

File: neptune/new/internal/artifacts/file_hash.py (hunk 2 above)
Comment (Raalsky) [reply neptune/new/internal/artifacts/file_hash.py]: No, I don't. It should have more than 3 stars :wink:

**Review by pkasprzyk — APPROVED** [review]

**Pull request merged**